### PR TITLE
Avoid DYNPRO_SEND_IN_BACKGROUND while ADT Pull

### PR DESCRIPTION
**Pull request by mfsap**
When ADT (aka ABAP in Eclipse) is used to trigger a pull request, there is no SAPGUI available in the backend system. Whenever a dynpro is called, a DYNPRO_SEND_IN_BACKGROUND short dump occurs.
This commit fixes issue #2632.

Additional remark: In case of a private repository, username and password are passed to the backend system. The ADT backend coding currently handles this via class ZCL_ABAPGIT_DEFAULT_AUTH_INFO. To avoid sending a password dialog in the backend system, class CL_ABAPGIT_PASSWORD_DIALOG needs to be adjusted to take over user/pw from ZCL_ABAPGIT_DEFAULT_AUTH_INFO in a generic way. I know this is not nice and it would be better to have ZCL_ABAPGIT_DEFAULT_AUTH_INFO in ZABAPGIT. Even the best solution would be to re-use ZCL_ABAPGIT_LOGIN_MANAGER instead, but I would recommend to refactor this in a new pull request later on...

### CODE REVIEW

**Comment by larshp** [comment]
regarding passwords, see https://github.com/abapGit/ADT_Backend/issues/21

File: src/ui/zcl_abapgit_password_dialog.clas.abap [hunk 1]
```diff
@@ -22,10 +22,39 @@ CLASS ZCL_ABAPGIT_PASSWORD_DIALOG IMPLEMENTATION.
 
   METHOD popup.
 
-    PERFORM password_popup
-      IN PROGRAM (sy-cprog)
-      USING iv_repo_url
-      CHANGING cv_user cv_pass.
+    DATA lv_gui_is_available TYPE abap_bool.
```
Comment (larshp) [inline src/ui/zcl_abapgit_password_dialog.clas.abap]: suggest moving this part to a different pull request

**Comment by larshp** [comment]
did you figure out the steps to reproduce the error from #2632 ?

**Comment by mfsap** [comment]
We had the same error with RS_CORR_INSERT in our ADT development landscape as well. It usually happens if the current user has no open task in the selected transport request and the backend system needs confirmation that a new task is created for the user. In case SUPPRESS_DIALOG in RS_CORR_INSERT, the task is automatically created w/o user interaction.

File: src/ui/zcl_abapgit_password_dialog.clas.abap (hunk 1 above)
Comment (mfsap) [reply src/ui/zcl_abapgit_password_dialog.clas.abap]: I‘ll try to remove this class form the current PR and create a new PR for this 'intermediate' solution soon...

File: src/objects/zcl_abapgit_objects_super.clas.abap [hunk 2]
```diff
@@ -164,7 +164,6 @@ CLASS ZCL_ABAPGIT_OBJECTS_SUPER IMPLEMENTATION.
         devclass            = iv_package
         master_language     = mv_language
         global_lock         = abap_true
-        author              = sy-uname
```
Comment (larshp) [inline src/objects/zcl_abapgit_objects_super.clas.abap]: I think the `suppress_dialog`s are safe to add, but I'm not sure about removing `author`

Comment (mfsap) [reply src/objects/zcl_abapgit_objects_super.clas.abap]: Adding the author was a fault by myself done by #2392. If you check all other `RS_CORR_INSERT` calls, the author is never provided.

Comment (larshp) [reply src/objects/zcl_abapgit_objects_super.clas.abap]: 👍
